kernel/reduction.ml

```diff
@@ -925,6 +925,17 @@ let hnf_prod_app env t n =
 let hnf_prod_applist env t nl =
   List.fold_left (hnf_prod_app env) t nl
 
+let hnf_prod_applist_assum env n c l =
+  let rec app n subst t l =
+    if Int.equal n 0 then
+      if l == [] then substl subst t
+      else anomaly (Pp.str "Not enough arguments.")
```

Comment:
Isn't this rather too many arguments?

Reply:
Well spotted. The error actually also appears twice in `term.ml`. I fixed all of them.